core/proguard-rules.pro

```diff
@@ -0,0 +1,25 @@
+##---------------Begin: proguard configuration for Gson  ----------
+# Took the congif from: https://github.com/google/gson/blob/master/examples/android-proguard-example/proguard.cfg
+
+# Gson uses generic type information stored in a class file when working with fields. Proguard
+# removes such information by default, so configure it to keep all of it.
+-keepattributes Signature
+
+# For using GSON @Expose annotation
+-keepattributes *Annotation*
+
+# Gson specific classes
+-dontwarn sun.misc.**
+
+# Application classes that will be serialized/deserialized over Gson
+-keepclassmembers enum com.kin.ecosystem.** { *; }
```

Comment:
sdk module should have -keepclassmembers enum com.kin.ecosystem.** { *; } if still needed (need to double check) core needs (if it really need it) 

-keepclassmembers enum com.kin.ecosystem.core.** { *; }